napari/layers/vectors/vectors.py

```diff
@@ -622,13 +622,23 @@ def edge_contrast_limits(
     @property
     def _view_face_color(self) -> np.ndarray:
         """(Mx4) np.ndarray : colors for the M in view vectors"""
+
+        # Create as many colors as there are visible vectors
         face_color = self.edge_color[self._view_indices]
         face_color[:, -1] *= self._view_alphas
```

Comment:
I'm noticing now: doesn't this `*=` actually change inplace the array in `color_manager`? I think this should be a copy maybe? @andy-sweet what do you think?

Reply:
I think `Vectors._view_indices` is an arbitrary list or `ndarray`, so the line here uses [fancy/advanced indexing](https://numpy.org/doc/stable/user/basics.indexing.html#advanced-indexing), which means we get a new array rather than a view of the underlying one in `ColorManager`.

That would actually be a very useful implementation comment to add here.

Reply:
Ah, that makes sense!

Reply:
I added a few lines about that.